ddtrace/internal/ci_visibility/_api_client.py

```diff
@@ -0,0 +1,521 @@
+class ITRData:
+    correlation_id: t.Optional[str] = None
+    covered_files: t.Optional[t.Dict[str, CoverageLines]] = None
+    skippable_items: t.Set[t.Union[InternalTestId, TestSuiteId]] = dataclasses.field(default_factory=set)
+
+
+class _SkippableResponseMeta(TypedDict):
+    coverage: t.Dict[str, str]
+    correlation_id: str
+
+
+class _SkippableResponseDataItemAttributes(TypedDict):
+    name: str
+    suite: str
+    parameters: str
+    configurations: t.Dict[str, t.Any]
+
+
+class _SkippableResponseDataItem(TypedDict):
+    type: str
+    attributes: _SkippableResponseDataItemAttributes
+
+
+class _SkippableResponse(TypedDict):
+    data: t.List[_SkippableResponseDataItem]
+    meta: _SkippableResponseMeta
+
+
+def _get_test_id_from_skippable_test(
+    skippable_test: _SkippableResponseDataItem, ignore_parameters: bool
+) -> InternalTestId:
+    test_type = skippable_test["type"]
+    if test_type != TEST:
+        raise ValueError(f"Test type {test_type} is not expected test type {TEST}")
+    module_id = TestModuleId(skippable_test["attributes"]["configurations"]["test.bundle"])
+    suite_id = TestSuiteId(module_id, skippable_test["attributes"]["suite"])
+    test_name = skippable_test["attributes"]["name"]
+    test_parameters = None if ignore_parameters else skippable_test["attributes"].get("parameters")
+    return InternalTestId(suite_id, test_name, test_parameters)
+
+
+def _get_suite_id_from_skippable_suite(skippable_suite: _SkippableResponseDataItem) -> TestSuiteId:
+    suite_type = skippable_suite["type"]
+    if suite_type != SUITE:
+        raise ValueError(f"Test type {suite_type} is not expected test type {SUITE}")
+
+    module_id = TestModuleId(skippable_suite["attributes"]["configurations"]["test.bundle"])
+    return TestSuiteId(module_id, skippable_suite["attributes"]["suite"])
+
+
+def _parse_covered_files(covered_files_data: t.Dict[str, str]) -> t.Optional[t.Dict[str, CoverageLines]]:
+    covered_files = {}
+    parse_errors = 0
+    for covered_file, covered_lines_bytes in covered_files_data.items():
+        try:
+            covered_lines = CoverageLines.from_bytearray(bytearray(b64decode(covered_lines_bytes)))
+            covered_files[covered_file] = covered_lines
+        except:  # noqa: E722
+            log.debug("Failed to parse coverage data for file %s", covered_file)
+            parse_errors += 1
+            continue
+
+    if parse_errors > 0:
+        log.warning("Failed to parse %d coverage files", parse_errors)
+
+    return covered_files
+
+
+def _parse_skippable_suites(
+    skippable_suites_data: t.List[_SkippableResponseDataItem],
+) -> t.Set[_SKIPPABLE_ITEM_ID_TYPE]:
+    suites_to_skip: t.Set[_SKIPPABLE_ITEM_ID_TYPE] = set()
+    count_unparsed_suites = 0
+    for skippable_suite in skippable_suites_data:
+        try:
+            suite_id = _get_suite_id_from_skippable_suite(skippable_suite)
+            suites_to_skip.add(suite_id)
+        except Exception:  # noqa: E722
```

Comment:
Good point. I think I just lack enough of an opinion to pick one. :D Maybe just `except:` ?

Reply:
Well, the code quality thingy seems to complain about bare `except:` but not about `except Exception:` (but then again, it may be updated to complain about `except Exception:` too at some point, so this is not a strong argument). One potential point in favor of `except Exception` is that it makes you stop and thing "do I really need to catch every exception?" (but then again, in the tracer we often do want that).

Reply:
> One potential point in favor of except Exception is that it makes you stop and thing "do I really need to catch every exception?" 

Doesn't `except:` make you stop and think "do I really need ignore everything?"?